src/lib/libc_preload/syscall_wrappers.c

```diff
@@ -137,12 +137,8 @@ INTERPOSE(epoll_pwait2);
 #ifdef SYS_epoll_wait // kernel entry: num=232 func=sys_epoll_wait
 INTERPOSE(epoll_wait);
 #endif
-#ifdef SYS_eventfd // kernel entry: num=284 func=sys_eventfd
-INTERPOSE(eventfd);
-#endif
-#ifdef SYS_eventfd2 // kernel entry: num=290 func=sys_eventfd2
-INTERPOSE(eventfd2);
-#endif
+// Skipping SYS_eventfd
+// Skipping SYS_eventfd2
```

Comment:
Has this function ever been defined in libc? Leaving it in wouldn't hurt much unless someone defines their own eventfd2 function and then we would end intercepting it.

Ultimately, I think it comes down to: Do we really want to maintain this special case? Maybe we don't want to maintain the `gen_syscall_wrappers_c.py` script at all and we should just delete it. Then we could just maintain the raw `syscall_wrappers.c` file directly without trying to make sure the script will regenerate it correctly.

(The reason I committed the script was that I thought it would be useful for us later to detect when new syscalls are added, but I don't think we necessarily need to use it to keep `syscall_wrappers.c` exactly up to date.)

Reply:
> Has this function ever been defined in libc? Leaving it in wouldn't hurt much unless someone defines their own eventfd2 function and then we would end intercepting it.

Glibc has never had an `eventfd2` function. I think it would be better to avoid generating wrappers that don't exist in glibc so that we don't accidentally cause unexpected behaviour in the plugin, but maybe we don't need to worry about it seriously.

> Ultimately, I think it comes down to: Do we really want to maintain this special case? Maybe we don't want to maintain the `gen_syscall_wrappers_c.py` script at all and we should just delete it. Then we could just maintain the raw `syscall_wrappers.c` file directly without trying to make sure the script will regenerate it correctly.
> 
> (The reason I committed the script was that I thought it would be useful for us later to detect when new syscalls are added, but I don't think we necessarily need to use it to keep `syscall_wrappers.c` exactly up to date.)

I don't think it's that bad to have special cases. We can remove the `gen_syscall_wrappers_c.py` script if you want, I don't have an opinion either way.

Reply:
OK, let's keep it for now. Thanks for the updates!